repair/row_level.cc

```diff
@@ -2990,3 +3041,31 @@ future<> repair_service::stop() {
 repair_service::~repair_service() {
     assert(_stopped);
 }
+
+future<std::optional<gc_clock::time_point>>
+repair_service::update_history(utils::UUID repair_id, utils::UUID table_id, dht::token_range range, gc_clock::time_point repair_time) {
+    auto shard = unsigned(repair_id.get_most_significant_bits()) % smp::count;
+    return container().invoke_on(shard, [repair_id, table_id, range, repair_time] (repair_service& rs) mutable {
+        repair_history& rh = rs._finished_ranges_history[repair_id];
+        if (rh.repair_time > repair_time) {
+            rh.repair_time = repair_time;
+        }
+        auto finished_shards = ++(rh.finished_ranges[table_id][range]);
+        if (finished_shards == smp::count) {
+            // All shards have finisehd repair the range. Send an rpc to ask peers to update system.repair table
+            rlogger.debug("repair[{}]: Finished range {} for table {} on all shards, updating system.repair table, finished_shards={}", repair_id, range, table_id, finished_shards);
+            return make_ready_future<std::optional<gc_clock::time_point>>(std::optional<gc_clock::time_point>(rh.repair_time));
+        } else {
+            rlogger.debug("repair[{}]: Finished range {} for table {} on all shards, updating system.repair table, finished_shards={}", repair_id, range, table_id, finished_shards);
+            return make_ready_future<std::optional<gc_clock::time_point>>(std::nullopt);
```

Comment:
and here: `co_return std::nullopt;` suffices.

Reply:
done.